vllm/v1/engine/detokenizer.py

```diff
@@ -101,135 +219,76 @@ def add_request(
         assert ((self.stream_mode and stream is not None)
                 or (not self.stream_mode and stream is None))
 
-        request_state = DetokenizerRequestState.from_new_request(
+        request_state = IncrementalDetokenizer.from_new_request(
             self.tokenizer, request, stream)
         self.request_states[request.request_id] = request_state
 
     def step(
-            self, encore_core_outputs: List[EngineCoreOutput]
-    ) -> List[RequestOutput]:
+        self, encore_core_outputs: List[EngineCoreOutput]
+    ) -> Tuple[List[RequestOutput], List[str]]:
         """Update state and request the RequestOutputs to the LLMEngine."""
 
         assert not self.stream_mode
 
         request_outputs: List[RequestOutput] = []
+        requests_to_abort: List[str] = []
         for engine_core_output in encore_core_outputs:
             request_id = engine_core_output.request_id
+            detokenizer = self.request_states[request_id]
```

Comment:
@njhill I believe this can result in a `KeyError` sometimes. This can happen when the LLMEngineCore runs ahead before it gets to process the abort request. What do you think ?

Reply:
@varun-sundar-rabindranath good catch! I think all that's needed here though is to ignore the case where we no longer have an entry (the request is already completed from our pov and we can discard residual outputs from the engine).